Source/Core/Core/AchievementManager.cpp

```diff
@@ -270,6 +272,75 @@ void AchievementManager::ActivateDeactivateRichPresence()
       nullptr, 0);
 }
 
+void AchievementManager::FetchBadges()
+{
+  if (!m_is_runtime_initialized || !IsLoggedIn() || !Config::Get(Config::RA_BADGES_ENABLED))
+    return;
+  if (!m_player_badge.loaded)
+  {
+    m_image_queue.EmplaceItem([this] {
+      {
+        std::lock_guard lg{m_lock};
```

Comment:
This block seems to be repeated in each of those cases.  Can you break it out into common function?

Reply:
Well, there's different inputs each time, but if you really want me to I can look into it.